R/Scores.R

```diff
@@ -115,62 +115,37 @@ setGeneric("evaluate", function(s, design, ...) standardGeneric("evaluate"))
 #' @export
 setMethod("evaluate", signature("IntegralScore", "TwoStageDesign"),
     function(s, design, optimization = FALSE, subdivisions = 10000L, ...) {
-        # use design-specific implementation
-        if (optimization) return(.evaluate(s, design, ...))
-        # use generic approach
         epsilon <- sqrt(.Machine$double.eps)
-        n1      <- n1(design)
-        early_futility <- predictive_cdf(s@data_distribution, s@prior, design@c1f, n1) *
-            evaluate(s@cs, design, design@c1f - epsilon) # score is constant on early stopping region
-        early_efficacy <- (1 - predictive_cdf(s@data_distribution, s@prior, design@c1e, n1)) *
-            evaluate(s@cs, design, design@c1e + epsilon)
-        continuation <- stats::integrate(
-            function(x1) predictive_pdf(s@data_distribution, s@prior, x1, n1, ...) * evaluate(s@cs, design, x1, ...),
-            lower        = design@c1f,
-            upper        = design@c1e,
-            subdivisions = subdivisions
-        )$value
-        return(early_futility + continuation + early_efficacy)
-    })
-
-
-
-
-
-# not user facing
-setGeneric(".evaluate", function(s, design, ...) standardGeneric(".evaluate"))
-
-
-
-# not user facing!
-setMethod(".evaluate", signature("IntegralScore", "TwoStageDesign"),
-    function(s, design, ...) {
-        epsilon <- sqrt(.Machine$double.eps)
-        n1 <- design@n1; c1f <- design@c1f; c1e <- design@c1e
+        c1f     <- design@c1f
+        c1e     <- design@c1e
+        n1      <- if (optimization) design@n1 else design@n1 # todo: this is wrong ;)
```

Comment:
@MatheMax this is the painpoint: we did not use rounded n1 in evaluate(optimization = FALSE) before!